referenceqvm/unitary_generator.py

```diff
@@ -355,6 +356,45 @@ def tensor_gates(gate_set, defgate_set, pyquil_gate, num_qubits):
     return gate
 
 
+def tensor_up(pauli_terms, num_qubits):
+    """
+    Takes a PauliSum object along with a total number of
+    qubits and returns a matrix corresponding the tensor representation of the
+    object.
+
+    Useful for generating the full Hamiltonian after a particular fermion to
+    pauli transformation.
+
+    :param pauli_terms: (PauliSum) object of PauliTerm
+    :param num_qubits: (int) number of qubits in the system
+    :returns: (numpy array) representation of the paui_terms operator
+    """
+    if not isinstance(pauli_terms, PauliSum):
+        raise TypeError("can only tensor PauliSum")
+
+    if __debug__:
```

Comment:
do we need to check in debug code?

Reply:
under default mode of running python (without -O), then this check occurs.  If run with -O the code is trusting the user to not send any PauliSums with non-identity terms on qubits with higher than the maximum.  Do you think the check should run for all mode's of operation? or just when the user expects safe and sane execution.

Reply:
I think this check should always occur. It's cheap as the `kron`ing operation is not executed in a loop over-and-over again if the circuit code is produced in a reasonable way.